tests/unit_tests/include/test_helper.hpp

```diff
@@ -53,6 +53,10 @@
             EXPECT_EQ(res, test_passed); \
     } while (0);
 
+#define CHECK_DOUBLE_ON_DEVICE(d)                                        \
+    if (d->get_info<sycl::info::device::double_fp_config>().size() == 0) \
+    GTEST_SKIP()
```

Comment:
Consider adding a message to explain why the test is skipped like so:

```suggestion
    GTEST_SKIP() << "Double precision is not supported on the device";
```